packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -1289,22 +1333,56 @@ export function updateProperties(
         const lastProp = lastProps[propKey];
         if (
           nextProps.hasOwnProperty(propKey) &&
-          nextProp !== lastProp &&
           (nextProp != null || lastProp != null)
         ) {
           switch (propKey) {
+            case 'type': {
+              type = nextProp;
+              // Fast path since 'type' is very common on inputs
```

Comment:
Changing 'type' is very uncommon though. setProp would be fewer bytes?

Reply:
I want to change this to `domElement.type = nextProp` but I leave that for a later refactor as well as dropping `function` and `symbol` as special cases which would then bring this down. For now I'm just highlighting that these special cases apply often. `setProp` would obscure that this should actually be simple.